homeassistant/components/esphome/entry_data.py

```diff
@@ -31,6 +31,8 @@
 
 DATA_KEY = "esphome"
 
+SAVE_DELAY = 120
```

Comment:
Is this going to be okay? If there never will be 2 minutes of no data, it will never save until HA is shut down.

Reply:
Should be fine since it only needs the data to re-setup the config entries with the last state before its able to make the connection to the device.

Reply:
For other use cases, we could consider adding a `max_wait` to `delay_save` to make sure it's written eventually (besides on shutdown).